GoedeDoelenHelpen/ClientApp/src/app/event/default-event-register/default-event-register.component.ts

```diff
@@ -0,0 +1,188 @@
+    });
+    this.charityC = this.charityForm.get('charity');
+    this.charities = [
+      {
+        name: 'WNF',
+        logo: 'https://www.designboom.com/wp-content/uploads/2015/11/panda-transormation-designboom-818.gif'
+      },
+      {
+        name: 'WEM',
+        logo: 'https://s3-eu-west-1.amazonaws.com/votecompany/5910900-www.goudengans.nl.6128438.jpg'
+      },
+      {
+        name: 'KIKA',
+        logo: 'https://i1.wp.com/10emeidoorn.nl/wp-content/uploads/2016/03/Kika-logo.jpg'
+      },
+      {
+        name: 'Wakker Dier',
+        logo: 'https://d2l8h2eumdtj2d.cloudfront.net/app/uploads/2018/01/Wakker-Dier-Logo-272x205.jpg'
+      }
+    ];
+    this.filteredCharities = this.charityC.valueChanges
+      .pipe(
+        startWith(''),
+        map(charity => charity ? this.filterCharities(charity) : this.charities.slice())
+      );
+
+    // event description
+    this.eventDescriptionForm = this.fb.group({
+      eventDescription: ['', Validators.compose([Validators.required, Validators.maxLength(280)])]
+    });
+    this.eventDescriptionC = this.eventDescriptionForm.get('eventDescription');
+
+    // event name
+    this.eventNameForm = this.fb.group({
+      eventName: ['', Validators.compose([Validators.required, Validators.minLength(4), Validators.maxLength(12)])]
+    });
+    this.eventNameC = this.eventNameForm.get('eventName');
+
+    // event date
+    this.eventDateForm = this.fb.group({
+      eventDate: ['', Validators.required]
+    });
+    this.eventDateC = this.eventDateForm.get('eventDate');
+
+    // profile image
+    this.profileImageForm = this.fb.group({
+      profileImage: ['']
+    });
+    this.profileImageC = this.profileImageForm.get('profileImage');
+
+    // password
+    this.passwordForm = this.fb.group({
+        password: [
+          '', [
+          Validators.compose([
+            Validators.required,
+            Validators.minLength(8),
+            Validators.maxLength(64),
+            // Validators.pattern('^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[A-Z])')
+          ])]
+        ],
+        passwordRepeat: ['', Validators.required]
+      }
+    );
+    this.passwordC = this.passwordForm.get('password');
+    this.passwordRepeatC = this.passwordForm.get('passwordRepeat');
+  }
+
+  filterCharities(name: string): Charity[] {
+    return this.charities.filter(charity =>
+      charity.name.toLowerCase().indexOf(name.toLowerCase()) === 0);
+  }
+
+  toLocalDate(dateString: Date) {
+    const options = { month: 'long', day: 'numeric' };
+    const date = new Date(dateString);
+    return date.toLocaleDateString('nl-NL', options);
+  }
+
+  validPassword(password: AbstractControl) {
+    return new RegExp('(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[A-Z])').test(password.value);
+  }
+
+  equalPasswords(password: AbstractControl, repeat: AbstractControl): boolean {
+    return password.value === repeat.value ? true : false;
+  }
+
+  allRequiredFieldsValid() {
+    if (
```

Comment:
you could als create a group of groups so you can use the built in api of FormGroup https://angular.io/api/forms/FormGroup